gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/UserQuotaManager.java

```diff
@@ -159,6 +179,9 @@ public boolean releaseQuota(Dag.DagNode<JobExecutionPlan> dagNode) {
       String proxyUserKey = DagManagerUtils.getUserQuotaKey(proxyUser, dagNode);
       decrementQuotaUsage(proxyUserToJobCount, proxyUserKey);
     }
+    String flowGroup = ConfigUtils.getString(dagNode.getValue().getJobSpec().getConfig(),
+        ConfigurationKeys.FLOW_GROUP_KEY, "");
+    decrementQuotaUsage(flowGroupToJobCount, DagManagerUtils.getFlowGroupQuotaKey(flowGroup, dagNode));
     String serializedRequesters = DagManagerUtils.getSerializedRequesterList(dagNode);
     if (serializedRequesters != null) {
```

Comment:
When should `releaseQuota` be called other than for the tests below to do manual testing? Is there a time period during which a quota is enforced ie: 5 concurrent jobs allowed for group/user X and then we call `releaseQuota` as flows finish executing?

Reply:
I have a test from a previous PR in`DagManagerTest` that also tested for releaseQuota API in terms of the larger flow lifecycle.

`releaseQuota` is only called when a flow is completed so besides that the only scenario where the count decrements is when an increment fails due to exceeding the quota, but this is tested in the below test  `testUserAndFlowGroupQuotaMultipleUsersAdd` since a failed increment is followed with a decrement then increment